pkg/storages/s3/session.go

```diff
@@ -231,9 +232,53 @@ func createSession(bucket string, settings map[string]string) (*session.Session,
 			}
 		})
 	}
+
+	if encodedHeaders, ok := settings[RequestAdditionalHeaders]; ok {
+		headers, err := getHeaders(encodedHeaders)
+		if err != nil {
+			return nil, err
+		}
+
+		s.Handlers.Validate.PushBack(func(request *request.Request) {
+			for k, v := range headers {
+				request.HTTPRequest.Header.Add(k.(string), v.(string))
+			}
+		})
+	}
+
 	return s, err
 }
 
+func getHeaders(encodedHeaders string) (map[interface{}]interface{}, error) {
```

Comment:
HTTP header keys and values known to be strings. I do not see a point for interface{} thing.